lms/djangoapps/verify_student/models.py

```diff
@@ -1189,3 +1190,28 @@ class Meta:
 
     def __str__(self):
         return str(self.arguments)
+
+
+class VerificationAttempt(TimeStampedModel):
+    """
+    The model represents impelementation-agnostic information about identity verification (IDV) attempts.
+
+    Plugins that implement forms of IDV can store information about IDV attempts in this model for use across
+    the platform.
+    """
+    user = models.ForeignKey(User, db_index=True, on_delete=models.CASCADE)
+    name = models.CharField(blank=True, max_length=255)
+
+    STATUS_CHOICES = [
+        VerificationAttemptStatus.created,
+        VerificationAttemptStatus.pending,
+        VerificationAttemptStatus.approved,
+        VerificationAttemptStatus.denied,
+    ]
+    status = models.CharField(max_length=64, choices=[(status, status) for status in STATUS_CHOICES])
+
+    expiration_datetime = models.DateTimeField(
+        null=True,
+        blank=True,
+        db_index=True,
+    )
```

Comment:
If your most common query is something like, "does this user have an unexpired verification?", you might want to explicitly make a composite index for that (`["user", "-expiration_datetime"]`). MySQL on RDS isn't always great about combining the intersection of two indexes, and can sometimes do a lot more work than it needs to.

Reply:
I took a look at the the query patterns on the IDV tables, and they're almost always querying on the `user` field and then sometimes on the `status` field. I didn't see any examples of querying on the `expiration_datetime`; the rows are sorted in Python. I removed the index on that field and left the index on the `user` field.

I considered adding a composite index on `["user", "status"]`, but I wasn't sure if that the overhead of maintaining the index would be worth the benefits of the index, since the number of `VerificationAttempts` per learner should be relatively small. I wish I could see the query plan for MySQL with and without that index but couldn't figure out how to do that.

If you have any thoughts about an index like this, let me know. I'll merge this PR as-is, but I'm happy to put in a follow up PR to adjust the index. Thanks!